### PR TITLE
Remove some dead IO code in System.Runtime.WindowsRuntime

### PR DESCRIPTION
**Pull request by stephentoub**
Found by a coverity scan
cc: @ianhays

### CODE REVIEW

File: src/System.Runtime.WindowsRuntime/src/System/IO/UnmanagedMemoryStreamInternal.cs [hunk 1]
```diff
@@ -226,10 +222,6 @@ public override long Position
             {
                 if (!_isOpen) throw new ObjectDisposedException(null, SR.ObjectDisposed_StreamClosed);
 
-                // Note: subtracting pointers returns an Int64.  Working around
-                // to avoid hitting compiler warning CS0652 on this line.
-                if (new IntPtr(value - _mem).ToInt64() > UnmanagedMemStreamMaxLength)
-                    throw new ArgumentOutOfRangeException("offset", SR.ArgumentOutOfRange_UnmanagedMemStreamLength);
```
Comment (ianhays) [inline src/System.Runtime.WindowsRuntime/src/System/IO/UnmanagedMemoryStreamInternal.cs]: A quick search shows this exception message only being used in this file. We should remove it from the SR file if that's the case.

**Comment by ianhays** [comment]
LGTM minus SR comment.

File: src/System.Runtime.WindowsRuntime/src/System/IO/UnmanagedMemoryStreamInternal.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime.WindowsRuntime/src/System/IO/UnmanagedMemoryStreamInternal.cs]: Good catch.  Removed.

**Comment by stephentoub** [comment]
Thanks, Ian.

**Pull request merged**